zebra-utils/src/bin/search-issue-refs/main.rs

```diff
@@ -0,0 +1,265 @@
+                    Some(ext) if check_file_ext(ext) => Some(vec![path]),
+                    _ => None,
+                }
+            } else {
+                None
+            }
+        })
+        .flatten()
+        .collect())
+}
+
+fn github_issue_url(issue_id: &str) -> String {
+    format!("https://github.com/ZcashFoundation/zebra/issues/{issue_id}")
+}
+
+fn github_remote_file_ref(file_path: &str, line: usize) -> String {
+    let file_path = &file_path[2..];
+    format!("https://github.com/ZcashFoundation/zebra/blob/main/{file_path}#L{line}")
+}
+
+fn github_issue_api_url(issue_id: &str) -> String {
+    format!("https://api.github.com/repos/ZcashFoundation/zebra/issues/{issue_id}")
+}
+
+#[derive(Debug)]
+struct PossibleIssueRef {
+    file_path: String,
+    line_number: usize,
+    column: usize,
+    id: String,
+}
+
+/// Process entry point for `search-issue-refs`
+#[allow(clippy::print_stdout, clippy::print_stderr)]
+#[tokio::main]
+async fn main() -> Result<()> {
+    init_tracing();
+    color_eyre::install()?;
+
+    let file_paths = search_directory(&".".into())?;
+
+    // Zebra's github issue numbers could be up to 4 digits
+    let issue_regex =
+        Regex::new(r"(https://github.com/ZcashFoundation/zebra/issues/|#)(\d{1,4})").unwrap();
+
+    let mut possible_issue_refs: Vec<PossibleIssueRef> = vec![];
+
+    for file_path in file_paths {
+        let file = File::open(&file_path)?;
+        let lines = io::BufReader::new(file).lines();
+
+        for (line_idx, line) in lines.into_iter().enumerate() {
+            let line = line?;
+
+            possible_issue_refs.extend(issue_regex.captures_iter(&line).map(|captures| {
+                let file_path = file_path
+                    .to_str()
+                    .expect("paths from read_dir should be valid unicode")
+                    .to_string();
+
+                let potential_issue_ref = captures.get(2).expect("matches should have 2 captures");
+                let matching_text = potential_issue_ref.as_str();
+
+                let id =
+                    matching_text[matching_text.len().checked_sub(4).unwrap_or(1)..].to_string();
+
+                PossibleIssueRef {
+                    file_path,
+                    line_number: line_idx + 1,
+                    column: captures
+                        .get(1)
+                        .expect("matches should have 2 captures")
+                        .start()
+                        + 1,
+                    id,
+                }
+            }))
+        }
+    }
+
+    let num_possible_issue_refs = possible_issue_refs.len();
+
+    println!(
+        "\nFound {num_possible_issue_refs} possible issue refs, checking Github issue statuses..\n"
+    );
+
+    // check if issues are closed on Github
+
+    let github_token = match env::vars().find(|(key, _)| key == GITHUB_TOKEN_ENV_KEY) {
+        Some((_, github_token)) => github_token,
+        _ => {
+            println!(
```

Comment:
```suggestion
            tracing::warn!(
```

Will not work as is, will need to bring `tracing` but i think that can make it a bit clearer that something didn't went as expected. I am happy to approve without this change, let me know.

Reply:
I'd like to avoid adding the dependency.

Reply:
sure.